### DISCUSSION TITLE
Is it possible to communicate with metamask from outside the browser?

### DISCUSSION THREAD

**Discussion by fasteater** [MetaMask/snaps#1100] [Q&A] [answered]
I wonder if there is any solution to communicate with metamask from outside of the browser, say from a local desktop app? Is this something snaps could do?

**Comment by ritave** [comment, marked as answer]
Currently no existing solution like this.

What you could do is request `endowment:internet-access` permission and use WebSockets to communicate with a local server.

**Reply by fasteater** [reply]
does that mean I need to run a local server?

**Reply by ritave** [reply]
Sadly, yes.
